doc/concept/build_system.md

```diff
@@ -0,0 +1,73 @@
+_Note: This document is designed to render nicely with GitHub-flavored Markdown.  My apologies in advance to everyone else._
+
+Build System Concepts
+=====================
+
+Goals of this document
+----------------------
+
+Be deleberate about choosing a build system; explain criteria, options, choices, and decisions.
+
+Strategy to choose a build system is:
+1. Enumerate good experiences to retain
+2. Enumerate bad experiences to avoid
+3. Be judiciously terse; the longer this is, the fewer people will read it
+4. Enumerate & experiment with options
+5. Make decision
+
+
+Definitions
+-----------
+
+**Contributor:** Who is a contributor?  Is there some base-level knowledge we can expect?
+
+
+Requirements
+------------
+
+* Minimal dependencies as practical, to facilitate easy building by others
+* Cross-platform
+* Any external dependencies must be freely installable (i.e., no registration, no license codes, no money; see also `CONTRIBUTING.md`)
+* Any internally satisfied dependencies must be freely runable (i.e., no license codes; see also `CONTRIBUTING.md`)
+* Is maintainable by contributors given a reasonable learning curve
+
+
+Good experiences to retain
```

Comment:
I think one experience we might want to support is meeting the users where they are. If users are already using a particular package manager, it would be good for us to support them there.

In particular, if CMake is the overwhelming majority, we should at least support CMake. If Bazel is next in the list, we should have some support for it too. Whether we test and/or keep both in-sync is another question, but I think we can support more than one if we're clear about what the priorities are and who will be owning which ones we're supporting.

Reply:
Sounds like you think a survey is a good idea.  I'll put one together.